code/modules/virus2/effect.dm

```diff
@@ -377,6 +377,37 @@
 	speech.message = message
 
 
+/datum/disease2/effect/vitreous
+	name = "Vitreous resonance"
+	stage = 2
+	chance = 25
+	max_chance = 75	
+	max_multiplier = 2
+
+datum/disease2/effect/vitreous/activate(var/mob/living/carbon/mob)
+	if(istype(mob, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = mob
+		var/hand_to_use = pick(GRASP_RIGHT_HAND, GRASP_LEFT_HAND)
+		var/obj/item/weapon/reagent_containers/glass_to_shatter = H.get_held_item_by_index(hand_to_use)
+		var/datum/organ/external/glass_hand = H.find_organ_by_grasp_index(hand_to_use)
+		if (istype(glass_to_shatter, /obj/item/weapon/reagent_containers/glass/) || istype(glass_to_shatter, /obj/item/weapon/reagent_containers/syringe))
+			to_chat(H, "<span class='warning'>Your [glass_hand.display_name] resonates with the glass in the [glass_to_shatter.name], shattering it to bits!</span>")
+			glass_to_shatter.reagents.reaction(get_turf(H), TOUCH)
```

Comment:
Well personally I'm against implicit ejection on deletion and I think deletion should always delete contents recursively. But it's sort of needed because we have no proper context management for "destruction" of objects in general.